README.md

```diff
@@ -6,24 +6,85 @@
 
 flannel is a virtual network that gives a subnet to each host for use with container runtimes.
 
-Platforms like Kubernetes assume that each container (pod) has a unique, routable IP inside the cluster.
-The advantage of this model is that it reduces the complexity of doing port mapping.
+Platforms like Kubernetes assume that each container (pod) has a unique, routable IP inside the cluster. The advantage of this model is that it reduces the complexity of doing port mapping.
 
-## Theory of Operation
+## How it works
+
+Flannel runs an agent, `flanneld`, on each host and is responsible for allocating a subnet lease out of a preconfigured address space. Flannel uses [etcd][etcd] to store the network configuration, allocated subnets, and auxiliary data (such as host's IP). Packets are forwarded using one of several [backend mechanisms][backends].
```

Comment:
Not sure if this needs to be changed here, but for accuracy, when running with the kube-subnet-mgr (yes it needs a better name), flanneld doesn't actually allocate the subnet lease